resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -64,7 +63,7 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
                     if (numAddresses > 0) {
                         // if there are multiple addresses: we shall pick one by one
                         // to support the round robin distribution
-                        promise.setSuccess(inetAddresses.get(index.getAndIncrement() % numAddresses));
+                        promise.setSuccess(inetAddresses.get(shiftSize(numAddresses)));
```

Comment:
The name is misleading, should be called something like `randomIndex`, I honestly don't understand why so much waste of CPU on generating a random number, the other solution is so much simple and better.